executor/union_scan.go

```diff
@@ -286,30 +339,39 @@ func (us *UnionScanExec) rowWithColsInTxn(t table.Table, h int64, cols []*table.
 	if err != nil {
 		return nil, err
 	}
-	v, _, err := tables.DecodeRawRowData(us.ctx, t.Meta(), h, cols, value)
+	colIDs := make(map[int64]int)
+	for i, col := range us.columns {
+		colIDs[col.ID] = i
+	}
+	return decodeRowData(us.ctx, us.table.Meta(), us.columns, colIDs, h, []byte{}, value)
+}
+
+func (us *UnionScanExec) getMemRow(h int64) ([]types.Datum, error) {
+	data, err := us.rowWithColsInTxn(us.table, h)
+	if err != nil {
+		return nil, err
+	}
+	mutableRow := chunk.MutRowFromTypes(us.retTypes())
```

Comment:
no need `chunk.MutRowFromTypes(us.retTypes())` for every miss index handle, better be `us` member and reuse it

Reply:
done. Thanks.